### ISSUE TITLE
"Couldn’t find `prisma.yml` file. Are you in the right directory?"

### ISSUE THREAD

**Issue by jferrettiboke** [graphcool/prisma#2146]
Hello!

I am experiencing an error when I run my tests with Jest, `yarn test`. The error is "Couldn’t find `prisma.yml` file. Are you in the right directory?".

```js
// tests/auth.test.js

// ...

beforeAll(async () => {
  await resetDatabaseData();
});

// ...
```

Basically, `resetDatabaseData()` executes this command: `prisma reset --env-file .env --force`.

```js
const { exec } = require("child_process");
const path = require("path");

function resetDatabaseData() {
  return new Promise((resolve, reject) => {
    exec(
      "prisma reset --env-file .env --force",
      {
        cwd: path.resolve(__dirname, "..")
      },
      (err, stdout, stderr) => {
        if (err) {
          return reject(err);
        }

        return resolve(stdout);
      }
    );
  });
}

module.exports = { resetDatabaseData };
```

The `resetDatabaseData()` function is working because I tested it creating a `script.js` file and executing `node tests/script.js`. I have tried a lot of things but nothing seems to work for me. Any idea?

**Comment by steve-a-jones** [comment]
See my related issue here: https://github.com/graphcool/prisma/issues/2126. I'm doing the exact same thing as you. 

There is low level prisma code that conditionally generates random directories when `NODE_ENV=test` (which is the case for testing). My hack for now is to overwrite the `NODE_ENV` environment var for the child process to anything but `test` so that I can perform a reset.

**Comment by marktani** [comment]
Thanks for chiming in @steve-a-jones, that does seem related. Can you confirm, @jferrettiboke?

**Comment by jferrettiboke** [comment]
Hi @steve-a-jones!

Thanks for your answer but it doesn't work for me. Here what I did. Let me know what's wrong.

```js
function resetDatabaseData() {
  // I rename it
  process.env.NODE_ENV = "https://github.com/graphcool/prisma/issues/2146";

  return new Promise((resolve, reject) => {
    exec(
      "prisma reset --env-file .env --force",
      {
        cwd: path.resolve(__dirname, "..")
      },
      (err, stdout, stderr) => {
        console.log("Exit code:", err);
        console.log("Program output:", stdout);
        console.log("Program stderr:", stderr);

        if (err) {
          return reject(err);
        }

        // I rename it again to its previous original value
        process.env.NODE_ENV = "test";

        return resolve(stdout);
      }
    );
  });
}
```

I run my tests but I am getting the same error. Any idea?

**Comment by steve-a-jones** [comment]
Hey @jferrettiboke! 

You're close, but the hack involves updating `process.env.NODE_ENV` on the child process itself (updating on the parent process itself had adverse affects for me)

Here's what works for me:

```javascript
   const child = execFile('yarn', [resetCmd], {
            cwd: process.cwd(), 
            env: {
                ...process.env,
                NODE_ENV:'not_test' //HACK!
            }
        }, function(e, stdout, stderr){...}
   });
```

**Comment by jferrettiboke** [comment]
@steve-a-jones Thanks! Everything is fine now.

**Comment by marktani** [comment]
Thanks a lot you two. I'll close this as it's resolved, please see #2126 for more background information.
